src/metabase/collections/api.clj

```diff
@@ -1381,12 +1399,24 @@
       (api/check-403 api/*is-superuser?*))
     ;; ok, go ahead and update it! Only update keys that were specified in the `body`. But not `parent_id` since
     ;; that's not actually a property of Collection, and since we handle moving a Collection separately below.
-    (let [updates (u/select-keys-when collection-updates :present [:name :description :authority_level])]
+    (let [updates (u/select-keys-when collection-updates :present [:name :description :authority_level :type])]
       (when (seq updates)
-        (t2/update! :model/Collection id updates)))
+        (t2/with-transaction [_conn]
+          (t2/update! :model/Collection id updates)
+          ;; if type is changing, cascade the type change to all child collections
+          (when (api/column-will-change? :type collection-before-update updates)
+            (let [child-location (collection/children-location collection-before-update)]
+              (t2/query-one {:update :collection
+                             :where [:like :location (str child-location "%")]
+                             :set {:type (:type updates)}}))
+            (when (= (:type updates) "remote-synced")
+              (collection/check-non-remote-synced-dependencies (t2/select-one :model/Collection :id id)))))))
     ;; if we're trying to move or archive the Collection, go ahead and do that
     (move-or-archive-collection-if-needed! collection-before-update collection-updates)
-    (events/publish-event! :event/collection-touch {:collection-id id :user-id api/*current-user-id*}))
+    (let [updated-collection (t2/select-one :model/Collection :id id)]
+      (when config/ee-available?
```

Comment:
is there a better way to do this? we have to because if a public-event! method isn't defined this will error.